### PR TITLE
Backport TriggerExampleTest fixes, Fix XMLSourceTest

### PR DESCRIPTION
**Pull request by tgroh**
TriggerExampleTest uses an equality method that does not match the assumptions
made by the test for qualities that make rows equal.

Use an explicit coder in XMLSourceTest to avoid having to make a change
to the underlying behavior of JAXBCoder that may be backwards incompatible
in some cases.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @bjchambers

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/io/XmlSourceTest.java [hunk 1]
```diff
@@ -227,6 +229,10 @@ public String toString() {
     }
   }
 
+
+  private Coder<Train> trainCoder =
+      AvroCoder.of(Train.class, ReflectData.AllowNull.get().getSchema(Train.class));
```
Comment (bjchambers) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/XmlSourceTest.java]: Comment on why we're using AvroCoder within an XmlSource test?

**Comment by bjchambers** [comment]
The description of the PR doesn't seem accurate. This is changing more than just TriggerExampleTest.

**Comment by tgroh** [comment]
Updated PR Description

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/io/XmlSourceTest.java (hunk 1 above)
Comment (tgroh) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/XmlSourceTest.java]: Done.

**Pull request merged**